_posts/2020-05-20-hive-acid.md

```diff
@@ -0,0 +1,317 @@
+---
+layout: post
+title:  "Hive ACID support in Presto"
+author: Shubham Tagra, Qubole
+excerpt_separator: <!--more-->
+---
+
+Hive ACID tables are supported in Presto since the 331 release. Hive ACID
+support is an important step towards GDPR/CCPA compliance and also towards Hive3
+support given it creates ACID tables by default.
+
+In this blog we will cover the concepts of Hive ACID tables along with the
+changes done in Presto to support them. We will also cover the performance tests
+on this integration and look at the future plans for this feature.
+
+<!--more-->
+
+# How to use Hive ACID tables in Presto
+
+Hive ACID tables are readable in Presto without any need to tweak configs, you
+only need to take care of these requirements:
+
+1. Use Presto version 331 or higher
+2. Use Hive 3 Metastore Server. Presto does not support old Hive2 ACID tables.
+
+Note that Presto cannot create or write to Hive ACID tables yet. You can create
+and write to Hive ACID tables via
+[Hive](https://cwiki.apache.org/confluence/display/Hive/Hive+Transactions)
+or [Spark](https://github.com/qubole/spark-acid) and use Presto to read these
+tables.
+
+# What are Hive ACID tables
+Hive ACID Tables, also called Transactional Tables in Hive, provide ACID
+semantics in Hive. This excerpt from
+[Hive documentation](https://cwiki.apache.org/confluence/display/Hive/Hive+Transactions)
+covers ACID traits well:
+> “ACID stands for four traits of database transactions:
+>Atomicity (an operation either succeeds completely or fails,
+>it does not leave partial data), Consistency (once an application performs an
+>operation the results of that operation are visible to it in every subsequent
+> operation), Isolation (an incomplete operation by one user does not cause
+>unexpected side effects for other users), and Durability (once an operation is
+>complete it will be preserved even in the face of machine or system failure).
+>These traits have long been expected of database systems as part of their
+>transaction functionality.“
+
+Several concepts like transactions, writeIds, deltas, locks, etc are added in
+Hive to achieve ACID semantics which we will cover later in the blog.
+
+# Need for Hive ACID Tables
+In any organisation, there is always a need to update or delete existing entries
+in tables e.g. a user writes or updates the review for the service availed a
```

Comment:
Grammar nit: add a comma after e.g.

"availed" is uncommon in American English and confused me when I first read the sentence. I'd switch it to "provided".